option.go

```diff
@@ -0,0 +1,172 @@
+// Modifications are Copyright 2022 CloudWeGo Authors.
+
+package csrf
+
+import (
+	"context"
+
+	"github.com/cloudwego/hertz/pkg/app"
+)
+
+// Option is the only struct that can be used to set Options.
+type Option struct {
+	F func(o *Options)
+}
+
+const (
+	csrfSecret     = "csrfSecret"
+	csrfSalt       = "csrfSalt"
+	csrfToken      = "csrfToken"
+	csrfHeaderName = "X-CSRF-TOKEN"
+
+	letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	// 6 bits to represent a letter index
+	letterIdBits = 6
+	// All 1-bits as many as letterIdBits
+	letterIdMask = 1<<letterIdBits - 1
+	letterIdMax  = 63 / letterIdBits
+)
+
+type CsrfNextHandler func(ctx context.Context, c *app.RequestContext) bool
+
+type CsrfExtractorHandler func(ctx context.Context, c *app.RequestContext) (string, error)
+
+// Options defines the config for middleware.
+type Options struct {
+	// Secret used to generate token.
+	//
+	// Default: csrfSecret
+	Secret string
+
+	// Ignored methods will be considered no protection required.
+	//
+	// Optional. Default: "GET", "HEAD", "OPTIONS", "TRACE"
+	IgnoreMethods []string
+
+	// Next defines a function to skip this middleware when returned true.
+	//
+	// Optional. Default: nil
+	Next CsrfNextHandler
+
+	// KeyLookup is a string in the form of "<source>:<key>" that is used
+	// to create an Extractor that extracts the token from the request.
+	// Possible values:
+	// - "header:<name>"
+	// - "query:<name>"
+	// - "param:<name>"
+	// - "form:<name>"
+	//
+	// Optional. Default: "header:X-CSRF-TOKEN"
+	KeyLookup string
+
+	// ErrorFunc is executed when an error is returned from app.HandlerFunc.
+	//
+	// Optional. Default: func(ctx context.Context, c *app.RequestContext) { panic(c.Errors.Last()) }
+	ErrorFunc app.HandlerFunc
+
+	// Extractor returns the csrf token.
+	//
+	// If set this will be used in place of an Extractor based on KeyLookup.
+	//
+	// Optional. Default will create an Extractor based on KeyLookup.
+	Extractor CsrfExtractorHandler
+}
+
+func (o *Options) Apply(opts []Option) {
+	for _, op := range opts {
+		op.F(o)
+	}
+}
+
+// OptionsDefault is the default options.
+var OptionsDefault = Options{
+	Secret: csrfSecret,
+	// Assume that anything not defined as 'safe' by RFC7231 needs protection
+	IgnoreMethods: []string{"GET", "HEAD", "OPTIONS", "TRACE"},
+	Next:          nil,
+	KeyLookup:     "header:" + csrfHeaderName,
+	ErrorFunc:     func(ctx context.Context, c *app.RequestContext) { panic(c.Errors.Last()) },
+}
+
+func NewOptions(opts ...Option) *Options {
+	options := &Options{
+		Secret:        OptionsDefault.Secret,
+		IgnoreMethods: OptionsDefault.IgnoreMethods,
+		Next:          OptionsDefault.Next,
+		KeyLookup:     OptionsDefault.KeyLookup,
+		ErrorFunc:     OptionsDefault.ErrorFunc,
+	}
+	options.Apply(opts)
+	return options
+}
+
+func WithSecret(secret string) Option {
+	return Option{
+		F: func(o *Options) {
+			o.Secret = secret
+		},
+	}
+}
+
```

Comment:
add comment  and unit test

Reply:
> add comment and unit test

Added comments, other than that I think the unit tests are complete enough that every option is tested.

Reply:
ok